### PR TITLE
[Refactor]Refactor sampler

### PR DESCRIPTION
**Pull request by wangxiyuan**
Refactor Sampler implementation from patch way to inherit from vLLM Sampler interface.

Next step: Make the op `TopKTopPSampler` in vLLM support custom ops register mechanism 

- vLLM version: v0.10.0
- vLLM main: https://github.com/vllm-project/vllm/commit/61a6905ab036fd00eafdb1b0ca130d5feccfe686

### CODE REVIEW

**Comment by wangxiyuan** [comment]
@Pr0Wh1teGivee please help review. Thanks.

**Comment by Pr0Wh1teGivee** [comment]
LGTM

File: tests/ut/sample/test_sampler.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import unittest
+from unittest import mock
+
+import torch
+
+from vllm_ascend.sample.sampler import AscendSampler, AscendTopKTopPSampler
+
+
+class TestAscendSampler(unittest.TestCase):
```
Comment (Yikun) [inline tests/ut/sample/test_sampler.py]: Better to use testbase to avoid testcase being impacted by other future patches

File: vllm_ascend/sample/sampler.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        logits: torch.Tensor,
+        k: torch.Tensor,
+        p: torch.Tensor,
+    ) -> torch.Tensor:
+        if p is not None and k is not None:
+            # npu_top_k_top_p's parameter order is (logits, p, k), not (logits, k, p)
+            return torch_npu.npu_top_k_top_p(logits, p, k)
+
+        probs = logits.softmax(dim=-1)
+        probs_sort, _ = probs.sort(dim=-1, descending=False)
```
Comment (Yikun) [inline vllm_ascend/sample/sampler.py:31]: Unrelated: it seems `probs` and `probs_sort` compution are ununsed `if p is None and k is None`, the logits can return directly.

File: tests/ut/sample/test_sampler.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+    def test_init_with_raw_logprobs(self):
+        sampler = AscendSampler(logprobs_mode="raw_logprobs")
+        self.assertEqual(sampler.logprobs_mode, "raw_logprobs")
+        self.assertTrue(hasattr(sampler, 'topk_topp_sampler'))
+        self.assertIsInstance(sampler.topk_topp_sampler, AscendTopKTopPSampler)
+
+
+class TestAscendTopKTopPSampler(unittest.TestCase):
```
Comment (Yikun) [inline tests/ut/sample/test_sampler.py]: same

**Review by Yikun — APPROVED** [review]

File: vllm_ascend/sample/sampler.py (hunk 2 above)
Comment (Yikun) [reply vllm_ascend/sample/sampler.py:31]: cc @Pr0Wh1teGivee

File: vllm_ascend/worker/model_runner_v1.py [hunk 4]
```diff
@@ -168,7 +168,15 @@ def __init__(self, vllm_config: VllmConfig, device: torch.device):
...
-        self.sampler = Sampler()
+        if envs.VLLM_ASCEND_ENABLE_TOPK_TOPP_OPTIMIZATION:
+            # TODO: drop the env config to use ascend sampler by default
+            from vllm_ascend.sample.sampler import AscendSampler
+
+            self.sampler = AscendSampler()
+        else:
+            from vllm.v1.sample.sampler import Sampler
+
+            self.sampler = Sampler()
```
Comment (weijinqian0) [inline vllm_ascend/worker/model_runner_v1.py:176]: Can it take effect directly by removing the flag?

Comment (wangxiyuan) [reply vllm_ascend/worker/model_runner_v1.py:176]: yes, I hope so. Let's have a fully test to ensure ascend sampler is stable enough. Then we can remove the flag

File: vllm_ascend/sample/sampler.py (hunk 2 above)
Comment (wangxiyuan) [reply vllm_ascend/sample/sampler.py:31]: Done

File: vllm_ascend/worker/model_runner_v1.py (hunk 4 above)
Comment (wangxiyuan) [reply vllm_ascend/worker/model_runner_v1.py:176]: I set the value to True to enable ascend sampler by default now.

**Review by Yikun — APPROVED** [review]
